### PR TITLE
[NATS] Fix issue in streaming publish

### PR DESCRIPTION
**Pull request by aashikam**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/20278

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/Utils.java [hunk 1]
```diff
@@ -64,11 +65,12 @@ public static Object bindDataToIntendedType(byte[] data, BType intendedType) {
         Object dispatchedData;
         switch (dataParamTypeTag) {
             case TypeTags.STRING_TAG:
-                dispatchedData = new String(data, StandardCharsets.UTF_8);
+                dispatchedData = new BmpStringValue(new String(data, StandardCharsets.UTF_8));
```
Comment (wggihan) [inline stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/Utils.java]: @manuranga Do we have other construct to create StringValue or is it ok to use this directly?

Comment (manuranga) [reply stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/Utils.java]: Don't directly use the constructor, please create a java string first and convert to BString using StringUtils.fromString.
BmpStringValue is an implimentation detail that may change. But we can't make it private becase it's used in genrated code.

Comment (aashikam) [reply stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/Utils.java]: Added this change in d156d8f5812d0781377f0636485f1b95c1772874 but still getting the issue https://github.com/ballerina-platform/ballerina-lang/issues/20306

**Comment by aashikam** [comment]
Blocked due to https://github.com/ballerina-platform/ballerina-lang/issues/20306

**Pull request merged**